crates/loader/src/local/assets.rs

```diff
@@ -63,21 +64,26 @@ impl FileMount {
 }
 
 /// Generate a vector of file mounts for a component given all its file patterns.
-pub fn collect(raw_mounts: &[RawFileMount], rel: impl AsRef<Path>) -> Result<Vec<FileMount>> {
+pub fn collect(
+    raw_mounts: &[RawFileMount],
+    exclude_patterns: &[glob::Pattern],
+    rel: impl AsRef<Path>,
+) -> Result<Vec<FileMount>> {
     let (patterns, placements) = uncase(raw_mounts);
 
-    let pattern_files = collect_patterns(&patterns, &rel)?;
-    let placement_files = collect_placements(&placements, &rel)?;
+    let pattern_files = collect_patterns(&patterns, exclude_patterns, &rel)?;
+    let placement_files = collect_placements(&placements, exclude_patterns, &rel)?;
```

Comment:
I'm a bit dubious about doing the filtering in both `collect_patterns` and `collect_placements`.  It significantly increases the complexity of those functions (which are already fairly complicated), and we seem to be doing very much the same filtering in both (although I may be missing some nuance here).

How viable is it to perform the filtering on the `all_files` collection, after we have done all the accumulation of files from patterns and placements?  For me this would be clearer if practical.

Reply:
Thanks for the suggestion. I remove `exclude_patterns` from `collect_patterns` and `collect_placements`. The function is simpler now.